packages/embark-proxy/src/index.ts

```diff
@@ -73,7 +73,7 @@ export default class ProxyManager {
 
     this.proxy = await new Proxy({events: this.events, plugins: this.plugins, logger: this.logger})
       .serve(
-        this.embark.config.blockchainConfig.endpoint,
+        node || this.embark.config.blockchainConfig.endpoint,
```

Comment:
Instead of ultimately getting the `--node` parameter value passed in to `proxy.serve`, could we not simply overwrite the `this.embark.config.blockchainConfig.endpoint` when the config is processed and only when in the test context? That way, we can avoid having to pass the `node` value around, and we ultimately know that we will be connecting the proxy to that endpoint (be it `http://localhost:8545` or `vm`).